packages/record-data/ember-cli-build.js

```diff
@@ -3,10 +3,23 @@
 const EmberAddon = require('ember-cli/lib/broccoli/ember-addon');
 
 module.exports = function(defaults) {
+  const isProd = process.env.EMBER_ENV === 'production';
+  const compatWith = process.env.EMBER_DATA_FULL_COMPAT ? '99.0' : null;
   let app = new EmberAddon(defaults, {
+    emberData: {
+      compatWith,
+    },
+    babel: {
+      // this ensures that the same build-time code stripping that is done
+      // for library packages is also done for our tests and dummy app
+      plugins: [...require('@ember-data/private-build-infra/src/debug-macros')(null, isProd, compatWith)],
+    },
     'ember-cli-babel': {
       throwUnlessParallelizable: true,
     },
+    'ember-cli-terser': {
+      exclude: ['assets/dummy.js', 'assets/tests.js', 'assets/test-support.js'],
+    },
```

Comment:
we may want to share a base ember-cli-build, testem, and test-helper.js file from our test-infra/build-infra package to reduce maintenance. Copied this over from ember-data for now.